src/lms-ds-loader/lms_ds_loader/arguments.py

```diff
@@ -0,0 +1,97 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+from lms_ds_loader.constants import Constants
+
+
+class Arguments:
+    """
+    Container for holding arguments parsed at the command line.
+
+    Parameters
+    ----------
+    csv_path : str
+        Base path for finding CSV files.
+    engine : str
+        Database engine, either "mssql" or "postgresql"
+    """
+
+    def __init__(self, csv_path, engine):
+        assert csv_path is not None, "csv_path cannot be None"
+        assert csv_path.strip() != "", "csv_path cannot be whitespace"
+        assert engine is not None, "engine cannot be None"
+        assert engine.strip() != "", "engine cannot be whitespace"

```

Comment:
If I convert this to a `dataclass`, then I'll lose my guard clauses, since __init__ will be generated automatically. There's a case to be made that I'm overdoing it with these guard clauses, but it will require someone swimming hard upstream to convince me of that case. I learned the hard way not to trust input data.

If I lose the guard clauses here, then I'll just have to guard against them elsewhere. But, since these values are not `readonly`, that might even be a better idea. After all, someone can always do this:

```
a = Arguments("some/path", "mssql")
a.engine = None

a.some_method()
```

And now if `some_method` were relying on `engine` not being `None`, then there might be an unexpected error. So `some_method` should really be checking for valid state of vars before using them. In which case checking them in the constructor was unnecessary.

I might have just talked myself into using the `@dataclass` attribute and improving the error handling closer to where it actually matters. I'll do that in the next story, the UPDATEs story.